app/controllers/facilities_controller.rb

```diff
@@ -29,9 +32,10 @@ def create
         phone_number: params[:facility][:phone_number],
         latitude: params[:facility][:longitude],
         longitude: params[:facility][:latitude],
-        type_code: params[:facility][:type_code],
```

Comment:
what happens when you say `params[:typename][0]`, then you wouldn't need to have that temporary `type_name_value` up there...